legion_core/src/world.rs

```diff
@@ -1877,7 +2268,7 @@ mod tests {
     }
 
     #[test]
-    fn merge() {
+    fn move_from() {
```

Comment:
Can we get some tests for `clone_from`?

Reply:
I'll look into doing something soon. A realistic usage requires some sort of type registry and an implementation for CloneImpl. The changes in this PR are the bare minimum, non-opinionated hooks required to allow someone to implement the functionality listed in the PR description in a reasonably efficient way.

Just to get a feel for how this ends up looking in a downstream crate, here's a higher-level implementation that allows using std library's Clone and From/Into traits, but also custom traits SpawnFrom/SpawnInto that allow the user to read components of other types and access systems.
https://github.com/kabergstrom/prefab/blob/master/legion-transaction/src/clone_merge.rs

In the simple case, if a component is cloneable or uses standard library's From/Into, the per-component code is really simple.

Here is a more complex case of the higher level layer being used:
https://github.com/aclysma/atelier-legion-demo/blob/master/demo/src/components/physics.rs

The source component type is a cloneable, serializable object that describes a circle or a box, like this:

```rust
pub struct RigidBodyBallComponentDef {
    pub radius: f32,
    pub is_static: bool,
}
```

The destination component type is a shape handle and a channel that lets us notify a system to destroy the shape

```rust
pub struct RigidBodyComponent {
    pub handle: DefaultBodyHandle,
    delete_body_tx: crossbeam_channel::Sender<DefaultBodyHandle>,
}
```

When the rigid body is spawned, the transform code is able to read the position, scale, and rotation components to properly place the body into the world.

So to return to your original comment, yes, I'd like to add some tests and even an example. But it will take a bit of time to do it and it may be better to wait until there is consensus on how the API should look. :)